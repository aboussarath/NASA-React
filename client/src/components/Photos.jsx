import React, { useState, useEffect } from 'react';

function Photos({ photos }) {

  return (
    <div className='photo-list'>
      {photos.map(photo => {
        return (
          <ul className='photo-detail' key={photo.id}>
            <li>Id: {photo.id}</li>
            <li>Camera: {photo.camera.name}</li>
            <a href={photo.img_src} target='_blank'>Click to view image</a>
          </ul>
        )
      })}
    </div>
  )
}

export default Photos;