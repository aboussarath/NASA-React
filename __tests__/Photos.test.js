import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Photos from '../client/src/components/Photos.jsx';
let mockData = require('../mock-photos.json');

configure({adapter: new Adapter()});

describe('Unit tests for Photos', () => {
  test('should render Photo list', () => {
    const photos = shallow(<Photos photos={mockData.photos}/>);

    expect(photos).toMatchSnapshot();
  })

  test('should be possible to click image links', () => {
    const photos = shallow(<Photos photos={mockData.photos}/>);

    photos.find('a.img-link').simulate('click');
  })
})


// Created test suite for App component, but issues with react-calendar