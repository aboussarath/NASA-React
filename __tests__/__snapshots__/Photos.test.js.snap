// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Photos should render Photo list 1`] = `ShallowWrapper {}`;
